include/kokkos-utils/callbacks/EventBeginEndIdMatcher.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_EVENTBEGINENDIDMATCHER_HPP
+#define KOKKOS_UTILS_CALLBACKS_EVENTBEGINENDIDMATCHER_HPP
+
+#include "kokkos-utils/callbacks/EventIdMatcher.hpp"
+#include "kokkos-utils/callbacks/Matcher.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+/**
+ * @brief Match a begin event and its corresponding end event based on their @c event_id.
+ *
+ * The @ref matcher can be used to match a specific begin event of type @p BeginEventType.
+ *
+ * @note The begin and end event types, constrained by the @ref BeginEvent and @ref EndEvent concepts,
+ *       respectively, have an @c event_id member. For a given event of a begin event type, there is
+ *       only one event of the corresponding end event type that will potentially have the same
+ *       @c event_id as the begin event. For instance, given a @ref BeginParallelForEvent
+ *       event, only one @ref EndParallelForEvent event should match.
```

Comment:
I have updated a bit the sentences. I removed the sentence that said that "Kokkos guarantees that ...". Because in the case of our manager, it's our manager that sets the event ids. And with `kokkos-tools`, it would be the tool that sets the event ids.